swift/common/middleware/s3api/acl_handlers.py

```diff
@@ -181,7 +190,124 @@ def get_acl(self, headers, body, bucket_owner, object_owner=None):
         return acl
 
 
-class BucketAclHandler(BaseAclHandler):
+class BucketPolicyHandler(BaseAclHandler):
+
+    def __init__(self, req, logger, container=None, obj=None, headers=None):
+        super(BucketPolicyHandler, self).__init__(req, logger, container, obj, headers)
+
+    def get_bucket_policy(self, body):
+        """
+        Get BucketPolicy instance from json body.
+        """
+        self.logger.debug("body %s", body)
+        self.logger.debug("body %s", type(body))
+        try:
+            import json
+            policy = BucketPolicy.from_dict(json.loads(body))
+        except Exception as e:
+            self.logger.error(e)
+            raise
+        return policy
+
+    def _handle_acl(self, app, sw_method, container=None, obj=None,
+                    permission=None, headers=None):
+        """
+        General bucket policy handling method.
+        This method expects to call Request._get_response() in outside of
+        this method so that this method returns response only when sw_method
+        is HEAD.
+        """
+        container = self.container if container is None else container
+        obj = self.obj if obj is None else obj
+        self.logger.debug("container %s", self.container)
+        self.logger.debug("obj %s", self.obj)
+        sw_method = sw_method or self.req.environ['REQUEST_METHOD']
+        resource = 'object' if obj else 'container'
```

Comment:
```suggestion
        resource = 'object' if obj is not None else 'container'
```
Otherwise `obj` can also be `''` which probably isn't correct. I think the above checks should be cleaned up. First decide if you are expecting any input parameters to have blank string. `None` and blank strings are not same.

Reply:
we already have check for none `obj = self.obj if obj is None else obj` . We are good here